File: app/src/main/java/com/example/animabuilder/SecondaryToggleClick.java
package com.example.animabuilder;

import android.view.View;
import android.widget.LinearLayout;
import android.widget.ToggleButton;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

public class SecondaryToggleClick implements View.OnClickListener {

    ToggleButton parent;
    Fragment toShow;
    FragmentManager fm;

    FragmentTransaction ft;

    public SecondaryToggleClick(ToggleButton parent, Fragment toShow, FragmentManager fm){
        this.parent = parent;
        this.toShow = toShow;
        this.fm = fm;
    }

    @Override
    public void onClick(View v) {
        ft = fm.beginTransaction();

        if(parent.isChecked())
            ft.show(toShow);
        else
            ft.hide(toShow);

        ft.commit();
    }
}

File: app/src/main/java/com/example/animabuilder/ListenerImplementations/SecondaryToggleClick.kt
package com.example.animabuilder.ListenerImplementations

import android.view.View
import android.widget.ToggleButton
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentTransaction

class SecondaryToggleClick(
    var parent: ToggleButton,
    var toShow: Fragment,
    var fm: FragmentManager
) : View.OnClickListener {
    var ft: FragmentTransaction? = null
    override fun onClick(v: View) {
        ft = fm.beginTransaction()
        if (parent.isChecked) ft!!.show(toShow) else ft!!.hide(toShow)
        ft!!.commit()
    }
}